tests/ip_api_test.go

```diff
@@ -0,0 +1,200 @@
+package tests
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/phoenixnap/go-sdk-bmc/ipapi"
+	"github.com/stretchr/testify/suite"
+)
+
+type IpApiTestSuite struct {
+	suite.Suite
+	Client *ipapi.APIClient
+	Ctx    context.Context
+}
+
+func (suite *IpApiTestSuite) SetupSuite() {
+	// Set configuration
+	configuration := ipapi.NewConfiguration()
+	configuration.Host = "127.0.0.1:1080"
+	configuration.Scheme = "http"
+
+	// Set the context background
+	suite.Ctx = context.WithValue(context.Background(), "accessToken", "ACCESSTOKENSTRING")
+	suite.Ctx = context.WithValue(suite.Ctx, "serverIndex", nil)
+
+	// New ApiClient
+	suite.Client = ipapi.NewAPIClient(configuration)
+}
+
+func (suite *IpApiTestSuite) TearDownTest() {
+	TestUtilsImpl{}.resetExpectations()
+}
+
+func (suite *IpApiTestSuite) VerifyCalledOnce(expectationId string) {
```

Comment:
imo this should be private unless we are extracting it to a separate file. It is only being called from within the suite and doesnt need to be called from other test file